src/util/cmdline.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Kernel cmdline parsing - utility functions
+
+use errors::*;
+use std::{fs, io};
+use std::io::prelude::*;
+
+// Get OEM ID value from cmdline file.
+pub fn get_oem(fpath: &str, flagname: &str) -> Result<String> {
+    // open the cmdline file
+    let file =
+        fs::File::open(fpath).chain_err(|| format!("Failed to open cmdline file ({})", fpath))?;
+
+    // read the contents
+    let mut bufrd = io::BufReader::new(file);
+    let mut contents = String::new();
+    bufrd.read_to_string(&mut contents)
+        .chain_err(|| format!("Failed to read cmdline file ({})", fpath))?;
+
+    match find_flag_value(flagname, contents) {
+        Some(s) => Ok(s),
+        None => bail!(
+            "Couldn't find flag '{}' in cmdline file ({})",
+            flagname,
+            fpath
+        ),
+    }
+}
+
+// Find OEM ID flag value in cmdline string.
+fn find_flag_value(flagname: &str, cmdline: String) -> Option<String> {
+    // split the contents into elements and keep key-value tuples only.
+    let params: Vec<(&str, &str)> = cmdline
+        .split(' ')
```

Comment:
`char::is_whitespace` ?

Reply:
`bootparam(7)` says `a list of strings (boot arguments) separated by spaces`. I don't think we should go for the UTF-8 correct way here.

Reply:
This will get mixed up with quoted spaces, right? E.g. `arg="my quoted string"`. Though feel free to file this under the "this is not a 100% compliant parser" ticket. :)

Reply:
Indeed, this doesn't have any kind of quoting/escaping awareness either (but I haven't checked if that kind of quoting is actually allowed, or backslash-escaping or no-spaces).